carmm/analyse/counterpoise_onepot.py

```diff
@@ -0,0 +1,161 @@
+def counterpoise_calc(complex_struc, a_id, b_id, symbol_not_index, fhi_calc=None, a_name='A', b_name='B',
+                      verbose=False, dry_run=False):
+    """
+    This function does counterpoise (CP) correction in one go, assuming a binding complex AB.
+    Let's say we have A and B in this complex
+    A_only has A in the geometry of the binding complex with its own basis
+    A_plus_ghost has A in the same geometry as in the complex with B replaced by ghost atoms
+    CP correction = A_only + B_only - A_plus_ghost - B_plus_ghost
+    This value should be added to the energy change of interest, such as adsorption energy.
+
+    Parameters:
+        complex_struc: ASE Atoms
+            This is the Atoms object which stores the optimized structure of the binding complex
+        a_id: list of atom symbols or atom indices for species A
+        b_id: list of atom symbols or atom indices for species B
+            Please use both symbols or both indices for a_id and b_id.
+        symbol_not_index: bool
+            This indicates whether symbols are used.
+        fhi_calc: ase.calculators.aims.Aims object
+            A FHI_aims calculator constructed with ase Aims
+        a_name: The name of the first species for your counterpoise correction, which has symbol (or index) a_id.
+        b_name: The name of the second species for your counterpoise correction, which has symbol (or index) b_id.
+        verbose: Flag for printing output.
+        dry_run: bool
+            Flag for test run (CI-test)
```

Comment:
It can be quite hard to replace the fhi_calc with a generic one. I give up the `calculate` function in `ase.calculators.calculator.FileIOCalculator.calculate` which is shared among most types of calculators in ase. I did this to be able to pass the `ghosts` kwarg to the write_input function explicitly. 
In ase 3.22.1, there is no explicit 'ghosts'  kwarg when creating an Aims calculator, so `ghosts` goes into `parameters` and get written into control.in instead of geometry.in.
If I do` fhi_calc.pop('ghosts')` before the calculation and set `fhi_calc.ghosts=*`, fhi_calc now has an `ghosts` attribute, but the calculation will still not work. That's because the call to `write_input` function inside the general `calculate` function,  `self.write_input(self.atoms, properties, system_changes)`, does not contain an `ghosts` kwarg.

Reply:
I can't get around this problem with ase 3.22.1 and have to write another `calculate` function which has `ghosts` kwarg in the call to `self.write_input`. By doing so, I can't use an EMT calculator as I will be calling an `write_input `function that does not exist for EMT.
This function could still work for other calculators providing they are using the same convention to write input files with ghost atoms as `write_aims` and their `write_input` function also accepts `ghosts` kwarg.

Reply:
I think this may be not that useful as other software (such as Gaussian?) may have counterpoise correction implemented already.
I know the Aims calculator and a few others are rewritten, and the problem described here is resolved in the new code. https://wiki.fysik.dtu.dk/ase/ase/calculators/calculators.html#calculators (in calculator configuration). This process would be much easier with the new code as one can supply a bool list for ghost atoms when creating the calculator. For now, these changes has not appear in current version of ase, so this function here is more a work around.

Reply:
OK, so we'll mark this as being an issue to return to when the new ASE release arrives.